tests/integration/test_hyper_transformer.py

```diff
@@ -101,110 +177,168 @@ def get_transformers():
     }
 
 
-def test_hypertransformer_with_transformers():
-    data = get_input_data_without_nan()
-    transformers = get_transformers()
-
-    ht = HyperTransformer(transformers)
-    ht.fit(data)
-    transformed = ht.transform(data)
-
-    expected = get_transformed_data()
+def test_hypertransformer_default_inputs():
+    """Test the HyperTransformer with default parameters.
 
-    np.testing.assert_allclose(
-        transformed.sort_index(axis=1).values,
-        expected.sort_index(axis=1).values
-    )
-
-    reversed_data = ht.reverse_transform(transformed)
-
-    original_names = data.pop('names')
-    reversed_names = reversed_data.pop('names')
+    This tests that if default parameters are provided to the HyperTransformer,
+    the ``default_transformers`` method will be used to determine which
+    transformers to use for each field.
 
-    pd.testing.assert_frame_equal(data.sort_index(axis=1), reversed_data.sort_index(axis=1))
+    Setup:
+        - `data_type_transformers` will be set to use the `CategoricalTransformer`
+        for categorical data types so that the output is predictable.
 
-    for name in original_names:
-        assert name not in reversed_names
+    Input:
+        - A dataframe with every data type.
 
-
-def test_hypertransformer_with_transformers_nan_data():
-    data = get_input_data_with_nan()
-    transformers = get_transformers()
-
-    ht = HyperTransformer(transformers)
+    Expected behavior:
+        - The transformed data should contain all the ML ready data.
+        - The reverse transformed data should be the same as the input.
+    """
+    data = get_input_data_without_nan()
+    expected_transformed = data.drop('datetime', axis=1)
+    expected_transformed.columns = [
+        'integer.value',
+        'float.value',
+        'categorical.value',
+        'bool.value',
+        'names.value'
+    ]
+    expected_transformed = get_transformed_data()
+    expected_reversed = get_reversed()
+
+    ht = HyperTransformer(data_type_transformers={'categorical': CategoricalTransformer})
     ht.fit(data)
     transformed = ht.transform(data)
 
-    expected = get_transformed_nan_data()
+    pd.testing.assert_frame_equal(transformed, expected_transformed)
 
-    np.testing.assert_allclose(
-        transformed.sort_index(axis=1).values,
-        expected.sort_index(axis=1).values
+    reversed = ht.reverse_transform(transformed)
+
+    pd.testing.assert_frame_equal(
+        expected_reversed.sort_index(axis=1),
```

Comment:
I kept thinking about this and did a couple tests, and I realized that the current behavior is not complete: If the entire table is transformed and then reverse transformed, the reversed data should be identical to the input data, including the columns order, which means that these `sort_index` calls should not be necessary.

I'll add a comment on the other PR about addressing this there, and then these tests can be updated